### PR TITLE
Bump rust from 1.55 to 1.58

### PR DESCRIPTION
**Pull request by feikesteenbergen**
The main benefit is that this includes support for [Rust 2021](https://doc.rust-lang.org/edition-guide/rust-2021/index.html) edition.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -244,6 +245,19 @@ RUN TS_VERSIONS="1.7.5 2.1.0 2.1.1 2.2.0 2.2.1 2.3.0 2.3.1 2.4.0 2.4.1 2.4.2 2.5
         /build/scripts/install_timescaledb.sh ${pg} ${TS_VERSIONS} || exit 1 ; \
     done
 
+# As we build many Rust Components, against many PostgreSQL versions, we really benefit from
+# a build cache, even if it only exists ephemerally in the container.
+RUN wget "https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz"
+RUN tar xzfp sccache*.tar.gz
```
Comment (itroot) [inline Dockerfile:251]: tar supports streams, so you can do something like  `wget -O - | tar xzfp`

File: Dockerfile [hunk 2]
```diff
@@ -348,6 +362,8 @@ RUN if [ ! -z "${TIMESCALEDB_TOOLKIT_EXTENSION}" -a -z "${OSS_ONLY}" ]; then \
 
 USER root
 
+RUN du -ms /build/cache
+
```
Comment (itroot) [inline Dockerfile:366]:
```suggestion
```

**Review by itroot — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (feikesteenbergen) [reply Dockerfile:251]: Yeah, will do, still playing with the exact bits here.

File: Dockerfile (hunk 2 above)
Comment (feikesteenbergen) [reply Dockerfile:366]: Yeah, WIP, will clean up

Comment (feikesteenbergen) [reply Dockerfile:366]: (Pushing some `wip` commits every so often as the purpose is to speed up CI/CD, this is to validate that assumption)

**Pull request closed without merging**